py/dml/types.py

```diff
@@ -477,20 +529,35 @@ def get_member_qualified(self, member):
         return t
 
     def cmp(self, other):
+        if self.const != other.const:
+            return NotImplemented
         if not other.is_int:
             return NotImplemented
+        if ((self.is_arch_dependent or other.is_arch_dependent)
+            and type(self) is not type(other)):
+            return NotImplemented
```

Comment:
Cleaner to express as `ArchDependentIntegerType.cmp`?

Reply:
Not quite good enough, consider `TInt.cmp(TLong)`. That said, maybe it's not unwarranted to just change the mechanism of how `IntegerType.cmp` works and make it per-subclass based, and get rid of the reliance on `is_int` and `is_endian`.

Reply:
Another option is to embrace that it is a cross-cutting concern, and make it a global function with `match` instead.

Reply:
aa

I think I'd hate that.

So: what are we doing for now? Status quo, or do you want me to be rid of the current architecture of `cmp` and `canstore` surrounding `IntegerType` and replace it with an entirely subclass-based one?